plugins/scaffolder-backend/src/scaffolder/stages/prepare/bitbucket.ts

```diff
@@ -18,35 +18,33 @@ import fs from 'fs-extra';
 import path from 'path';
 import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
 import { parseLocationAnnotation } from '../helpers';
-import { InputError, Git } from '@backstage/backend-common';
+import { Git } from '@backstage/backend-common';
 import { PreparerBase, PreparerOptions } from './types';
+import { BitbucketIntegrationConfig } from '@backstage/integration';
 import parseGitUrl from 'git-url-parse';
-import { Config } from '@backstage/config';
 
 export class BitbucketPreparer implements PreparerBase {
-  private readonly privateToken: string;
-  private readonly username: string;
-
-  constructor(config: Config) {
-    this.username =
-      config.getOptionalString('scaffolder.bitbucket.api.username') ?? '';
-    this.privateToken =
-      config.getOptionalString('scaffolder.bitbucket.api.token') ?? '';
+  static fromConfig(config: BitbucketIntegrationConfig) {
+    return new BitbucketPreparer(
+      config.username,
+      config.token,
+      config.appPassword,
+    );
   }
 
+  constructor(
+    private readonly username?: string,
+    private readonly token?: string,
+    private readonly appPassword?: string,
```

Comment:
this is one place where it's generally proven neat to just do `constructor(private readonly auth: { username?: string; ... }) {}`